File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        int largest=0;
        int slargest=0;
        while (n>0){
            int currdigit= n%10;
            if (currdigit>largest){
                slargest= largest;
                largest= currdigit;
            }
            else if (currdigit>slargest && currdigit<=largest){
                largest= largest;
                slargest= currdigit;
            }
            n=n/10;
        }
        return (largest*slargest);
    }
};

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        largest=0
        slargest=0
        while n>0:
            a= n%10
            if a>largest:
                slargest= largest
                largest=a
            elif a>slargest and a<=largest:
                largest=largest
                slargest=a
            n=n//10
        return (largest*slargest)